### PR TITLE
Makes doc methods display enums.

### PR DESCRIPTION
**Pull request by PJB3005**
I apologize for the redundant formatting changes, that was a formatting plugin I had installed.

Though seriously *there was a bloody semicolon in here*.

image

Also made `[enum Error]` link to an enum.

### CODE REVIEW

File: doc/tools/makerst.py [hunk 1]
```diff
@@ -156,7 +157,7 @@ def rstize_text(text, cclass):
 
     # Escape * character to avoid interpreting it as emphasis
     pos = 0
-    next_brac_pos = text.find('[');
```
Comment (PJB3005) [inline doc/tools/makerst.py]: 👀

File: doc/tools/makerst.py [hunk 2]
```diff
@@ -106,6 +106,7 @@ def make_class_list(class_list, columns):
 
     f.close()
 
+
```
Comment (mhilbrunner) [inline doc/tools/makerst.py]: duplicate newline that should be removed

File: doc/tools/makerst.py [hunk 3]
```diff
@@ -299,16 +301,27 @@ def make_type(t):
         return ':ref:`' + t + '<class_' + t.lower() + '>`'
     return t
 
+
```
Comment (mhilbrunner) [inline doc/tools/makerst.py]: duplicate newline that should be removed

File: doc/tools/makerst.py (hunk 2 above)
Comment (PJB3005) [reply doc/tools/makerst.py]: Now, I'm fine with removing it, though I do wanna mention that PEP8 states there should be two newlines between functions.

Comment (mhilbrunner) [reply doc/tools/makerst.py]: Ah, okay, ignore me then :P

**Review by mhilbrunner — APPROVED** [review]

**Pull request merged**